tools/vendor_pyprima.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+# Vendors pyprima from https://github.com/libprima/prima
+
+set -o nounset
+set -o errexit
+
+REPO_URL="https://github.com/libprima/prima"
+COMMIT_HASH="6fbf15d9b4aac20d2e3ee5b3fcb41f0ca17c4330"
+
+# XXX: run this from the repo top level like `./tools/vendor_pyprima.sh`
+ROOT_DIR="scipy/_lib/pyprima"
+
+rm -rf $ROOT_DIR
+mkdir $ROOT_DIR
+git clone $REPO_URL $ROOT_DIR
+cd $ROOT_DIR
+git checkout $COMMIT_HASH
+rm -rf .development
+rm -rf .git
+rm .gitmodules
+rm -rf benchmark/
+rm -rf fortran/
+rm -rf matlab/
+rm -rf c/
+rm -rf python/
+rm -rf .github/
```

Comment:
I think we should do something like this, it removes all but the pyprima directory and we don't have to worry so much about upstream reorganization breaking our script 
```suggestion
rm -rf $(ls -A | grep -v pyprima)
```

Reply:
I committed something slightly different in https://github.com/scipy/scipy/pull/22778/commits/22b64cff104241464b76ed8c9957e3fe557bb25c, let me know what you think